### PR TITLE
Demo app: Add carousel component

### PR DESCRIPTION
**Pull request by JohnSundell**
This change adds a carousel component to the demo app, partly for demonstration purposes - but also to enable us to cheaply perform UI tests and manual verification for a component with children (the demo app previously didn’t have any components with children).

The “Pretty pictures” feature of the demo app is also updated to display a carousel on the top, as well as adding tappable links to the pictures that it displays.

simulator screen shot 25 oct 2016 17 39 24

### CODE REVIEW

**Comment by JohnSundell** [comment]
@spotify/objc-dev

File: demo/sources/CarouselComponent.swift [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        return UIEdgeInsets(top: 0, left: horizontalMargin, bottom: 0, right: horizontalMargin)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        if let existingItemSize = itemSize {
+            return existingItemSize
+        }
+        
+        guard let model = self.model else {
+            return .zero
```
Comment (Goos) [inline demo/sources/CarouselComponent.swift]: Swift's elegance still catches me off-guard 😲

File: demo/sources/CarouselComponent.swift [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    func configureView(with model: HUBComponentModel, containerViewSize: CGSize) {
+        self.model = model
+        itemSize = calculateItemSize(forModel: model, containerViewSize: containerViewSize)
+        collectionView.reloadData()
+    }
+    
+    // MARK: - HUBComponentWithRestorableUIState
+    
+    func currentUIState() -> Any? {
+        return collectionView.contentOffset
```
Comment (Goos) [inline demo/sources/CarouselComponent.swift]: I'm a bit rusty on the swift/obj-c interop – in obj-c structs these structs would need to be boxed (as the method takes an `id`). Will this work?

Comment (JohnSundell) [inline demo/sources/CarouselComponent.swift]: Auto boxing FTW! 😍

Comment (Goos) [inline demo/sources/CarouselComponent.swift]: stop

**Pull request merged**